sql/core/src/main/scala/org/apache/spark/sql/GroupedDataset.scala

```diff
@@ -46,9 +55,16 @@ class GroupedDataset[K, T] private[sql](
       throw new UnsupportedOperationException("Only expression encoders are currently supported")
   }
 
+  /** Encoders for built in aggregations. */
+  private implicit def newLongEncoder: Encoder[Long] = ExpressionEncoder[Long](flat = true)
```

Comment:
can we import the implicit from `SQLImplicits`? e.g. `import queryExecution.sqlContext.implicits.newLongEncoder`